src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/resource/EligibilityAndSuitabilityResource.kt

```diff
@@ -0,0 +1,217 @@
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun getEligibilityCriterion(
+    @Parameter(required = true) @PathVariable prisonNumber: String,
+    @Parameter(required = true) @PathVariable code: String,
+  ): EligibilityCriterionView =
+    eligibilityAndSuitabilityService.getEligibilityCriterionView(prisonNumber, code)
+
+  @GetMapping("/offender/{prisonNumber}/current-assessment/suitability/{code}")
+  @PreAuthorize("hasAnyRole('ASSESS_FOR_EARLY_RELEASE_ADMIN')")
+  @Operation(
+    summary = "Returns a specific suitability check for a prisoner's current assessment",
+    description = "Returns details of a specific suitability for a prisoner's initial checks",
+    security = [SecurityRequirement(name = "assess-for-early-release-admin-role")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Returns details of a specific suitability criteria in the current assessment for the prisoner",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = SuitabilityCriterionView::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorised, requires a valid Oauth2 token",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Forbidden, requires an appropriate role",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class),
+          ),
+        ],
+      ),
+    ],
+  )
+  fun getSuitabilityCheck(
+    @Parameter(required = true) @PathVariable prisonNumber: String,
+    @Parameter(required = true) @PathVariable code: String,
+  ): SuitabilityCriterionView =
+    eligibilityAndSuitabilityService.getSuitabilityCriterionView(prisonNumber, code)
+
+  @PutMapping("/offender/{prisonNumber}/current-assessment/eligibility-and-suitability-check")
+  @PreAuthorize("hasAnyRole('ASSESS_FOR_EARLY_RELEASE_ADMIN')")
+  @Operation(
+    summary = "Sets the state of a current eligbility/suitability check",
+    description = "Returns details of a specific suitability for a prisoner's initial checks",
+    security = [SecurityRequirement(name = "assess-for-early-release-admin-role")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "204",
```

Comment:
should we return 200 as you're are returning a response?

Reply:
oh, this is a copy paste error, I'll fix in a follow up

Reply:
not it's not, it's just wrong, it should be a 204

Reply:
yes, not everyone agrees on these things but I've always used 204 for a put with an empty response.